dp-hello-world-event/event/consumer.go

```diff
@@ -0,0 +1,104 @@
+package event
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ONSdigital/dp-hello-world-event/schema"
+	kafka "github.com/ONSdigital/dp-kafka/v2"
+	"github.com/ONSdigital/log.go/log"
+)
+
+//go:generate moq -out mock/handler.go -pkg mock . Handler
+
+// MessageConsumer provides a generic interface for consuming []byte messages
+type MessageConsumer interface {
+	Channels() *kafka.ConsumerGroupChannels
+}
+
+// Handler represents a handler for processing a single event.
+type Handler interface {
+	Handle(ctx context.Context, helloCalled *HelloCalled) error
+}
+
+// Consumer consumes event messages.
+type Consumer struct {
+	closing chan bool
+	closed  chan bool
+}
+
+// NewConsumer returns a new consumer instance.
+func NewConsumer() *Consumer {
+	return &Consumer{
+		closing: make(chan bool),
+		closed:  make(chan bool),
+	}
+}
+
+// Consume converts messages to event instances, and pass the event to the provided handler.
+func (consumer *Consumer) Consume(ctx context.Context, messageConsumer MessageConsumer, handler Handler) {
+
+	go func() {
+		defer close(consumer.closed)
+
+		for {
```

Comment:
I would suggest changing this `Consume` with something similar to the consume() function in the dp-kafka examples: https://github.com/ONSdigital/dp-kafka/blob/main/examples/consumer-concurrent/main.go#L122

basically, we would create a loop that consumes from the Upstream channel, until it is closed:
```
for {
    consumedMessage, ok := <-upstream
    if !ok {
        break
    }
    ...
```